irobotclient/main.py

```diff
@@ -0,0 +1,100 @@
+"""
+Copyright (c) 2017 Genome Research Ltd.
+
+This program is free software: you can redistribute it and/or modify it
+under the terms of the GNU General Public License as published by the
+Free Software Foundation, either version 3 of the License, or (at your
+option) any later version.
+
+This program is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with this program. If not, see <http://www.gnu.org/licenses/>.
+"""
+import errno
+
+from os import path
+from requests import Response
+
+from irobotclient import configuration_handler
+from irobotclient import request_formatter
+from irobotclient.custom_exceptions import IrobotClientException
+from irobotclient.request_handler import Requester
+
+
+def _print_error_details(error: OSError):
+    """
+    Print the output of any errors raised.
+
+    Utilises the OSError exception class which the iRobot Client uses for custom exceptions too.
+
+    :param error:
+    :return:
+    """
+    print("Inside _output_error_details") # Beth - debug
+    print(error)
+    exit(error.errno)
+
+
+def _download_data(response: Response, output_dir:str):
+    """
+
+    :param response:
+    :param output_dir:
+    :return:
+    """
+    try:
+        file_name = (path.split(response.url))[1]
+
+        if response.headers["content-type"] == "application/octet-stream":
+            with open(f"{output_dir}{file_name}", "wb") as file:
+                file.write(bytes(response.content))
+        else:
+            with open(f"{output_dir}{file_name}", "w") as file:
+                file.write(response.content)
+    except OSError as err:
+        _print_error_details(err)
+    except:
+        raise IrobotClientException(errno=errno.ECONNABORTED, message="Cannot write content to file.")
+
+
+def _run():
```

Comment:
It looks to me like this is only designed to get one file per run (plus the index, automatically, if the `get_file_extensions` function returns a few). It would be nice if you could specify getting multiple files and your client did the right thing with regard to automatic index expansion.